prow/cmd/sub/main.go

```diff
@@ -130,6 +131,7 @@ func main() {
 		ConfigAgent:   configAgent,
 		Metrics:       promMetrics,
 		ProwJobClient: kubeClient,
+		Reporter:      reporter.NewReporter(configAgent.Config),
```

Comment:
Why does the `sub` component need to report at all? I thought that was handled by the crier pubsub reporter.

Reply:
We are reporting if we are unable to create the job for any given reason.

Reply:
We are reporting any issue of why we are unable to create a prow job. The only thing we can't report is when the payload cannot be parsed since we won't have destination to send it to.

Reply:
this report happens when we fail to create a prowjob :man_shrugging:

Reply:
(also the code was already there... this is fixing the initial nil reference)

Reply:
Hmmm -- in that case, won't we also need that in `plank` and `jenkins-operator` for when we fail to create `ProwJob`s? So we can't really break out the logic into `crier`?

Reply:
@stevekuznetsov plank and jenkins-operator doesn't create prowjobs. Hook may, but currently we swallow the error when user does `/test foo` when foo is invalid.

Reply:
Oh sorry lol wasn't thinking. In `hook` today do we just not report anything? What should we report on infrastructure errors? When we can't create a ProwJob that's like a comment saying please contact an admin, right? Not a job report

Reply:
yeah, if you look at https://github.com/kubernetes/test-infra/blob/master/prow/pubsub/reporter/reporter.go#L92-L118, the only place we need a reporter client is `message := c.generateMessageFromPJ(pj)`, we can extract out the actual reporting logic there so it won't require a real report client.

Reply:
let's address as a follow up